src/Applicator/AbstractApplicator.php

```diff
@@ -21,4 +23,46 @@ protected function escape(string $column, Value $value): Value
             ? $this->escape->escape($column, $value)
             : $value;
     }
+
+    public function getPreparedValue(FilterInterface $filter): array
+    {
+        $values = $filter->getValue()->getValue();
+
+        return is_iterable($values)
+            ? $this->getPreparedMultiValues($filter)
+            : $this->getPreparedSingleValue($filter);
+    }
+
+    protected function getPreparedMultiValues(FilterInterface $filter): array
+    {
+        $preparedValues = [];
+        $i = 0;
+        foreach ($filter->getValue()->getValue() as $value) {
+            $preparedValues[$this->createColumnPlaceholder('', $filter, (string) $i++)] = $value;
+        }
+
+        return $preparedValues;
+    }
+
+    private function createColumnPlaceholder(string $prefix, FilterInterface $filter, string ...$additional): string
+    {
+        $pieces = Seq::init(function () use ($filter, $additional) {
```

Comment:
Yes, its quite yagni now, but it is prepared for future pull request .. and I can't cover it now, because I dont want to expose this functionality outside..

Reply:
If it is prepared for future PR, the the best is to add it with the future PR ;). (With corresponding PR).

How should one review functionality, which is not being used anywhere neither covered by tests?

Reply:
It is now covered in the added test on multivalues.
https://github.com/lmc-eu/api-filter/pull/7/files#diff-927c05d8795daf3a84877ba79456acc2R60

Reply:
Ok, so now the method could be simply
```php
private function createColumnPlaceholder(string $prefix, FilterInterface $filter, string $additional = ''): string
{
    return $prefix . $filter->getColumn() . '_' . $filter->getTitle()
        . ($additional !== '' ? '_' . $additional : '');

}
```
or for better readability
```php
private function createColumnPlaceholder(string $prefix, FilterInterface $filter, string $additional = ''): string
{
    $placeholder = $prefix . $filter->getColumn() . '_' . $filter->getTitle();
    
    if ($additional !== '') {
        $placeholder .= '_' . $additional;
    }
    
    return $placeholder;
}
```

What is the benefit of increasing complexity by converting the `$additional` to traversable via `...` (while it cannot be used with more than one value) and of using the commonly unfamiliar and for this usecase also IMHO quite unneeded yield (why it is needed, if it is called exactly once?).

Reply:
I've refactored it into a `non-yield` version..

Reply:
:+1: Thanks, its much easier to understand the code now :). Its imho also a favor for any future maintainer or contributor of this library :).